pkg/apis/event_stats.go

```diff
@@ -2,19 +2,77 @@ package apis
 
 import (
 	"net/http"
+	"strconv"
+	"strings"
+	"time"
 
 	"github.com/gin-gonic/gin"
 	"github.com/harshvirani7/event-stats-test/model"
 	"github.com/harshvirani7/event-stats-test/pkg/cache"
 	"github.com/harshvirani7/event-stats-test/pkg/config"
+	"github.com/harshvirani7/event-stats-test/pkg/monitor"
 	"github.com/harshvirani7/event-stats-test/pkg/storage"
+	"github.com/harshvirani7/event-stats-test/utils"
+	"github.com/prometheus/client_golang/prometheus"
 	"go.uber.org/zap"
 )
 
+type RequestDetailCount struct {
+	StoreEventDataSuccessCnt               int `json:"storeEventDataSuccessCnt"`
+	TotalEventCountByTypeSuccessCnt        int `json:"totalEventCountByTypeSuccessCnt"`
+	TotalEventCountByCameraIdSuccessCnt    int `json:"totalEventCountByCameraIdSuccessCnt"`
+	EventCountSummaryByCameraIdSuccessCnt  int `json:"eventCountSummaryByCameraIdSuccessCnt"`
+	EventCountSummaryByEventTypeSuccessCnt int `json:"eventCountSummaryByEventTypeSuccessCnt"`
+	SummaryByCameraIdSuccessCnt            int `json:"summaryByCameraIdSuccessCnt"`
+	SummaryByEventTypeSuccessCnt           int `json:"summaryByEventTypeSuccessCnt"`
+}
+
+var ReqDetailCount RequestDetailCount
+
 type EventStats struct {
 	Logger    *zap.SugaredLogger
 	RdbClient *cache.Redis
 	Cfg       config.Config
+	Metrics   *monitor.Metrics
+}
+
+func init() {
+
+	ReqDetailCount = RequestDetailCount{
+		StoreEventDataSuccessCnt:               0,
+		TotalEventCountByTypeSuccessCnt:        0,
+		TotalEventCountByCameraIdSuccessCnt:    0,
+		EventCountSummaryByEventTypeSuccessCnt: 0,
+		EventCountSummaryByCameraIdSuccessCnt:  0,
+		SummaryByEventTypeSuccessCnt:           0,
+		SummaryByCameraIdSuccessCnt:            0,
+	}
+}
+
+// MonitoringMiddleware is a middleware function for monitoring HTTP requests.
+func MonitoringMiddleware(cfg config.Config, es EventStats) gin.HandlerFunc {
```

Comment:
This function can be moved to main.go file

Reply:
updated